### PR TITLE
dhcpv4: use offered ip in requested ip option

### LINKED ISSUE #302: DHCP: is setting address before ACKed ok?
Migrating my project to the now merged version of the DHCP client, I noticed that the IP address of the interface has to be set to the OFFERed address before the client will send out a proper REQUEST.  I feel like this is racy, and the "official" IP of the interface should only be set once the server has ACKed the request.

The client is already tracking state information in its `state` enum (which btw I think should be available publicly, or at least as a discriminant-only Discovering/Requesting/Renew), so the last offered IP could also go there?

/cc @astro

### PR DESCRIPTION
**Pull request by MabezDev**
Instead of configuring the ifaces ip address when a dhcp server offers
and ip, the client now use the requested_ip dhcp option, based on the offered ip
(yiaddr). The user now only has to configure the iface once the dhcp
transaction has completed (Ack'd).

Closes #302.

cc: @astro

### CODE REVIEW

**Comment by MabezDev** [comment]
@whitequark would it be possible to squeeze this in be fore the next release? Not sure who's best to review this as @astro is busy atm I think.

**Comment by whitequark** [comment]
Okay, I'll review it before that.

File: src/dhcp/clientv4.rs [hunk 1]
```diff
@@ -198,9 +198,8 @@ impl Client {
         };
         net_debug!("DHCP recv {:?} from {} ({})", dhcp_repr.message_type, src_ip, server_identifier);
 
-        let config = if (dhcp_repr.message_type == DhcpMessageType::Offer ||
-                         dhcp_repr.message_type == DhcpMessageType::Ack) &&
-                      dhcp_repr.your_ip != Ipv4Address::UNSPECIFIED {
+        // once we recieve the ack, we can pass the config to the user
```
Comment (astro) [inline src/dhcp/clientv4.rs]: s/recieve/receive/ :)

**Comment by MabezDev** [comment]
@whitequark I see this didn't make it into 0.6 :(, Will this now have to wait till 0.7 or could it make it into a patch release? (I don't *think* this is breaking change)

**Comment by whitequark** [comment]
We can make a patch release just fine. I was just really low on time lately--as soon as I review this, I can cut a release for you.

**Comment by MabezDev** [comment]
Thanks. I appreciate the time you've taken to review my PR's thus far. Let me know if there is anything I can do to help.

**Review by astro — APPROVED** [review]

**Pull request merged**